wp/lib/bap_wp/src/precondition.ml

```diff
@@ -1199,18 +1201,60 @@ let init_vars (vars : Var.Set.t) (env : Env.t) : Constr.t list * Env.t =
         debug "Initializing var: %s\n%!" (Constr.to_string comp);
         comp :: inits, env)
 
+let user_func_spec (sub_name : string) (sub_pre : string) (sub_post : string)
+    (sub : Sub.t) (_ : Arch.t) : Env.fun_spec option =
+  debug "Making user-defined subroutine spec with subroutine-name: %s, pre:
+%s, post: %s \n%!" sub_name sub_pre sub_post;
+  if String.equal sub_name (Sub.name sub) then
+    Some {
+      spec_name = "user_func_spec";
+      spec = Summary (fun env post tid ->
+          (* turn strings into proper smtlib2 statements; incr stack_ptr *)
+          let sub_pre : Constr.t = Z3_utils.mk_smtlib2_single env sub_pre in
```

Comment:
Small stylistic recommendation: it would be easier to read this if all of these lets were lifted above the record definition (and the `Some` constructor), say at the level of the `then` branch.

Reply:
That would look cleaner, I agree.